flake.nix

```diff
@@ -1,14 +1,19 @@
 {
-  outputs = { self, nixpkgs }: {
-    lib = {
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+        inherit (pkgs) callPackage;
+      in
+      {
+        # this does lock us to a different nixpkgs version for the docker build
+        # steps but it ensures the API is what we expect
+        lib.docker = callPackage ./docker.nix {};
```

Comment:
I think I'd put this in `utils` together with `oci` since it's system dependent. I think the convention is that `lib` is system independent.

Reply:
sounds right 👍